internal/gateway/queryengine.go

```diff
@@ -0,0 +1,98 @@
+package gateway
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	_ "github.com/marcboeker/go-duckdb"
+	"github.com/tablelandnetwork/sqlparser"
+)
+
+type QueryEngine struct {
+	db *sql.DB
+}
+
+func NewQueryEngine() (*QueryEngine, error) {
+	db, err := sql.Open("duckdb", "")
+	if err != nil {
+		return nil, fmt.Errorf("open duckdb: %s", err)
+	}
+
+	_, err = db.Exec("INSTALL https; LOAD https;")
+	if err != nil {
+		return nil, fmt.Errorf("install https: %s", err)
+	}
+
+	return &QueryEngine{
+		db: db,
+	}, nil
+}
+
+func (qe *QueryEngine) Query(ctx context.Context, sql string) (*TableData, error) {
+	ast, err := sqlparser.Parse(sql)
+	if err != nil {
+		return nil, err
+	}
+	ast.Statements[0].(*sqlparser.Select).From.(*sqlparser.AliasedTableExpr).Expr.(*sqlparser.Table).Name = sqlparser.Identifier(fmt.Sprintf("read_parquet(['http://34.106.97.87:8002/v1/os/t2gh7m2iaqvwv2oexsaetncmdm6hhac7k6ne3teda/%s'])", ast.Statements[0].(*sqlparser.Select).From.(*sqlparser.AliasedTableExpr).Expr.(*sqlparser.Table).Name.String()))
```

Comment:
replace the table name from SQL statement to `fmt.Sprintf("read_parquet(['http://34.106.97.87:8002/v1/os/t2gh7m2iaqvwv2oexsaetncmdm6hhac7k6ne3teda/%s']`